pkg/orchestration/wiring/kubeingress/plugin.go

```diff
@@ -232,15 +265,34 @@ func buildIngressHostName(resourceName voyager.ResourceName, sc wiringplugin.Sta
 		clusterHostPath)
 }
 
-func extractKubeComputeDependency(dependencies []wiringplugin.WiredDependency) (*apps_v1.Deployment, bool /* retriable */, error) {
+func extractSingleDependencyOfType(dependencies *[]wiringplugin.WiredDependency, resourceType voyager.ResourceType) (*wiringplugin.WiredDependency, bool /* retriable */, error) {
+	var matchedDependency *wiringplugin.WiredDependency
+	for _, dependency := range *dependencies {
+		if dependency.Type == resourceType {
+			if matchedDependency != nil {
+				return nil, false, errors.Errorf("must depend on a single %s resource, but multiple were found", resourceType)
+			}
+			matchedDependency = &dependency
```

Comment:
Ha-ha =) This will point to to the loop variable which changes on every iteration.
https://github.com/golang/go/wiki/Range
https://github.com/golang/go/wiki/CommonMistakes related